connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala

```diff
@@ -530,6 +530,24 @@ class SparkConnectClientSuite extends ConnectFunSuite with BeforeAndAfterEach {
     assert(reattachableIter.resultComplete)
   }
 
+  test("SPARK-48056: Client execute gets INVALID_HANDLE.SESSION_NOT_FOUND and proceeds") {
+    startDummyServer(0)
+    client = SparkConnectClient
+      .builder()
+      .connectionString(s"sc://localhost:${server.getPort}")
+      .enableReattachableExecute()
+      .build()
+    service.errorToThrowOnExecute = Some(new StatusRuntimeException(
+      Status.INTERNAL.withDescription("INVALID_HANDLE.SESSION_NOT_FOUND")))
+
+    val plan = buildPlan("select * from range(10000000)")
```

Comment:
nit: this test probably doesn't need such a large query, doing just a range(100) may maybe shave 50-100ms or sth, which is not a lot but adds up in thousands of tests...

Reply:
totally makes sense, I'll adjust the test a little bit.

Reply:
edit: even better, make it just a `select 1`, no need to launch any Spark jobs